src/main/java/subway/application/LineService.java

```diff
@@ -1,53 +1,73 @@
 package subway.application;
 
-import org.springframework.stereotype.Service;
-import subway.dao.LineDao;
-import subway.domain.Line;
-import subway.dto.LineRequest;
-import subway.dto.LineResponse;
+import static java.util.stream.Collectors.toList;
 
 import java.util.List;
-import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.domain.Line;
+import subway.domain.repository.LineRepository;
+import subway.domain.Section;
+import subway.dto.line.LineCreateRequest;
+import subway.dto.line.LineDetailResponse;
+import subway.dto.line.LineResponse;
+import subway.dto.line.LineUpdateRequest;
+import subway.dto.section.SectionResponse;
+import subway.exception.IllegalLineException;
 
 @Service
+@Transactional
 public class LineService {
-    private final LineDao lineDao;
+    private final LineRepository lineRepository;
 
-    public LineService(LineDao lineDao) {
-        this.lineDao = lineDao;
+    public LineService(LineRepository lineRepository) {
+        this.lineRepository = lineRepository;
     }
 
-    public LineResponse saveLine(LineRequest request) {
-        Line persistLine = lineDao.insert(new Line(request.getName(), request.getColor()));
-        return LineResponse.of(persistLine);
+    public LineResponse saveLine(LineCreateRequest request) {
+        Line line = new Line(request.getLineName(), request.getColor());
+        if (lineRepository.isDuplicateLine(line)) {
+            throw new IllegalLineException("해당 노선의 색, 또는 이름이 중복됩니다.");
+        }
+        long savedId = lineRepository.save(line);
+        return LineResponse.from(lineRepository.findById(savedId));
```

Comment:
다시 조회하기보다는 완벽한 객체를 만들어서 내려주는게 
비용을 아낄수 있을것 같아요!
```suggestion
        long savedId = lineRepository.save(line);
        return LineResponse.from(new Line(savedId, line.getName(), line.getColor()));
```

Reply:
저도 주어진 정보로 새롭게 객체를 만들어서 반환할지, 아니면 저장된 값을 조회해서 반환할지에 대해 고민을 많이 했었습니다 😢 

Repository에서 다시 조회하여 반환하는 방식을 선택한 이유는 **요청의 가장 끝이라고 볼 수 있는 데이터베이스에도 정상적으로 저장, 조회가 된다** 라는 것을 알려주기 위해서 였습니다!

영이께서 전달받은 값을 사용해 다시 객체를 생성해 반환하는 방법을 제안하신 이유가
"DB에는 정상적인 값이 저장될 것이라 생각하고, 요청된 값을 올바르게 받았다 라는 것을 나타내는 것 만으로도 충분하기 때문" 이라고 생각하면 될까요?

요청을 잘 받았고, 잘 처리했다 라는 것을 보여줄 책임은 있지만 DB에 어떻게 저장이 되었는가에 대한 책임은 없기 때문이라고 생각했는데, 영이의 의도와 맞는지 궁금합니다 ㅎㅎ

Reply:
객체가 제대로 잘 만들어졌다면 당연히 DB에는 저장될것이고
실패했다면 반환하지 않고 터질것이기 때문에 문제없을것 같아요!

Reply:
앗
 > 실패했다면 반환하지 않고 터질것이기 때문에

이 부분을 생각도 하지 않았네요ㅎㅎㅎ 😄 
영이의 말씀처럼 **객체가 잘 만들어졌다면 저장은 예외가 터지지 않는 이상 잘 이루어진 것**이 맞는 것 같습니다!!

감사합니다 😄